src/System.Runtime.WindowsRuntime/tests/System/IO/AsWinRTStreamTests.cs

```diff
@@ -0,0 +1,413 @@
+            Stream managedStream = TestStreamProvider.CreateReadOnlyStream();
+            using (IInputStream ins = managedStream.AsInputStream())
+            {
+                Assert.Same(managedStream, ins.AsStreamForRead());
+            }
+        }
+
+        [Fact]
+        public static void AsOutputStream_RoundtripUnwrap()
+        {
+            // NetFx Stream -> IOutputStream -> NetFx Stream -> roundtrip reference equality is preserved
+            Stream managedStream = TestStreamProvider.CreateWriteOnlyStream();
+            using (IOutputStream outs = managedStream.AsOutputStream())
+            {
+                Assert.Same(managedStream, outs.AsStreamForWrite());
+            }
+        }
+
+        [Fact]
+        public static void AsInputStream_Equal()
+        {
+            Stream stream = TestStreamProvider.CreateReadOnlyStream();
+
+            using (IInputStream insOne = stream.AsInputStream())
+            using (IInputStream insTwo = stream.AsInputStream())
+            {
+                Assert.Equal(insOne, insTwo);
+            }
+        }
+
+        [Fact]
+        public static void AsInputStream_NotEqual()
+        {
+            Stream streamOne = TestStreamProvider.CreateReadOnlyStream();
+            Stream streamTwo = TestStreamProvider.CreateReadOnlyStream();
+
+            Assert.NotEqual(streamOne, streamTwo);
+
+            using (IInputStream insOne = streamOne.AsInputStream())
+            using (IInputStream insTwo = streamTwo.AsInputStream())
+            {
+                Assert.NotEqual(insOne, insTwo);
+            }
+        }
+
+        [Fact]
+        public static void AsOutputStream_Equal()
+        {
+            Stream stream = TestStreamProvider.CreateWriteOnlyStream();
+
+            using (IOutputStream outsOne = stream.AsOutputStream())
+            using (IOutputStream outsTwo = stream.AsOutputStream())
+            {
+                Assert.Equal(outsOne, outsOne);
+            }
+        }
+
+        [Fact]
+        public static void AsOutputStream_NotEqual()
+        {
+            Stream streamOne = TestStreamProvider.CreateWriteOnlyStream();
+            Stream streamTwo = TestStreamProvider.CreateWriteOnlyStream();
+
+            Assert.NotEqual(streamOne, streamTwo);
+
+            using (IOutputStream outsOne = streamOne.AsOutputStream())
+            using (IOutputStream outsTwo = streamTwo.AsOutputStream())
+            {
+                Assert.NotEqual(outsOne, outsTwo);
+            }
+        }
+
+        [Fact]
+        public static void TestRead_MemoryStream_None()
+        {
+            DoTestRead(() => { return TestStreamProvider.CreateMemoryStreamAsInputStream(); }, InputStreamOptions.None, false, true);
+        }
+
+        [Fact]
+        public static void TestRead_MemoryStream_Partial()
+        {
+            DoTestRead(() => { return TestStreamProvider.CreateMemoryStreamAsInputStream(); }, InputStreamOptions.Partial, false, true);
```

Comment:
Nit: the return statement format here isn't necessary, could just be e.g.
```C#
DoTestRead(() => TestStreamProvider.CreateMemoryStreamAsInputStream, InputStreamOptions.Partial, false, true);
```
Also, it'd be helpful if those Boolean args could be named, to make it clear what they are at the call site, e.g.
```C#
DoTestRead(() => TestStreamProvider.CreateMemoryStreamAsInputStream, InputStreamOptions.Partial, mustInvokeProgressHandler:false, completesSynchronously:true);
```